lib/course_planner/accounts/users.ex

```diff
@@ -39,6 +39,14 @@ defmodule CoursePlanner.Accounts.Users do
     end
   end
 
+  def delete(id, current_user_id) do
+    if to_string(id) == to_string(current_user_id) do
+      {:error, :self_deletion}
+    else
+      delete(id)
+    end
+  end
+
   def delete(id) do
```

Comment:
maybe it is better to make this one a private function called `do_delete` so we don't get confused later

Reply:
done